### ISSUE TITLE
Add a new rule to prevent importing "package.json"

### ISSUE THREAD

**Issue by jakub-g** [import-js/eslint-plugin-import#1991]
# High level description

It's too easy to accidentally import whole `package.json` in the output bundle. 99% of the time, this is not what the person developing wants. I think this deserves a separate rule, even if `no-restricted-paths` could be used to enforce this.

# Why importing `package.json` is a bad idea?

1) It's a [security risk](https://medium.com/@alex.birsan/dependency-confusion-4a5d60fec610) when `package.json` ends up being public (a lot of internal data might be exposed, like names of internal packages, internal hostnames (from `scripts` or git URLs of git-based dependencies) etc. and even just exposing the info about public devDependencies which might not be desirable; while most of the devs would think it's safe to put those there, because "how could possibly package.json end up in the bundle?")

2) It bloats the bundle size. Typical `package.json` in a big app is multiple kilobytes of size.

# How it can happen?

Many reason, but for example: suppose you want to send  `version` of your app or some dependency as part of telemetry. A tempting solution is to just `require('../../package.json').version` or  `require('../../package.json').dependencies.foobar`.

Since webpack and other bundlers can read JSON files, it will just work™. However, as ES modules are indivisible (apart from exports which are tree-shakeable, but it's separate story), so it's not only `version` that ends up in the bundle, but _whole `package.json`._

(The safer solution would be to use something like [`DefinePlugin`](https://webpack.js.org/plugins/define-plugin/)).

**Comment by ljharb** [comment]
I don't agree it's a real security risk. That supply chain attack only is possible when an organization ignores the many-years-old best practice of having all internal npm packages be under a scope that the org publicly owns, and configuring their internal registry to not interact with the public registry for the internal scope name.

URLs are not secret information, and security by obscurity is no security at all.

As for bundle size, sure, but this could apply for any JSON import, no?

What are the use cases for pulling in a version number? I can imagine a few, but all of them are node-only use cases.

When would any tool that runs in the browser be looking at a `dependencies` list?
